cdptools/natural_language_analyzers/entity_analyzer.py

```diff
@@ -1,14 +1,311 @@
+    """
+    return {"start": span.start, "end": span.end, "label": label, "value": value}
+
+
 class EntityAnalyzer(NLAnalyzer):
+    def __init__(self):
+        self.NLP = en_core_web_sm.load()
+        self._entity_annotators_by_label = {
+            "PERSON": _annotate_person_entities,
+            "CARDINAL": _annotate_empty,
+            "DATE": _annotate_date_entities,
+            "ORG": _annotate_as_label("Entity[Organization]"),
+            "GPE": _annotate_as_label("Entity[GPE]"),
+            "ORDINAL": _annotate_empty,
+            "FAC": _annotate_as_label("Entity[FAC]"),
+            "LOC": _annotate_as_label("Entity[Location]"),
+            "TIME": _annotate_empty,
+            "NORP": _annotate_as_label("Entity[NORP]"),
+            "QUANTITY": _annotate_empty,
+            "PRODUCT": _annotate_product_entities,
+            "MONEY": _annotate_money_entities,
+            "PERCENT": _annotate_empty,
+            "LAW": _annotate_as_label("Entity[Law]"),
+            "EVENT": _annotate_as_label("Entity[Event]"),
+            "WORK_OF_ART": _annotate_as_label("Entity[WorkOfArt]"),
+            "LANGUAGE": _annotate_empty,
+        }
+
+    @staticmethod
+    def _create_doc(text, nlp_model):
+        """
+        Given a spacy NLP model, analyze text, returning a spacy doc.
+        """
+        return nlp_model(text)
+
+    def _annotate_entities(self, entities, doc, event_time):
+        """
+            Generates a list of annotations for each entity type in a list of entities.
+
+            Parameters
+            ----------
+            entities: List[spacy.tokens.Span]
+                A list of spacy.tokens.Span objects representing entities
+            doc: spacy.tokens.Doc
+                The spacy doc the entities are derived from
+            event_time: datetime
+                The timestamp that the event occurred on
+
+            Returns
+            -------
+            annotations
+                A dict of lists of Annotation objects, keyed by the annotation type
+        """
+        # Group the extracted entities by entity label
+        sorted_entities = sorted(entities, key=lambda e: e.label_)
+        grouped_entities = groupby(sorted_entities, key=lambda e: e.label_)
+        entities_by_label = {
+            entity_type: list(es) for entity_type, es in grouped_entities
+        }
+
+        # Generate the annotations as appropriate by type
+        annotations_by_entity_type = {}
+        for entity_type, es in entities_by_label.items():
+            entity_type_annotator = self._entity_annotators_by_label[entity_type]
+            annotations_by_entity_type[entity_type] = entity_type_annotator(
+                es, doc, event_time
+            )
+
+        return annotations_by_entity_type
+
     def load(self, transcript, event_metadata):
         sentences_texts = [sentence["text"] for sentence in transcript["sentences"]]
         return (sentences_texts, event_metadata["event_datetime"])
```

Comment:
Thinking about the general version of this, should we always expect to have to unpack the output of `NLAnalyzer.load` rather than having it return one object that `analyze` can accept not-unpacked? Originally I was hoping the resulting code would look more like:
```
input = EntityAnalyzer.load(event["transcript"], event["metadata"])
entities = EntityAnalyzer.analyze(input)
```
I do like that the arguments to `analyze` make the inputs more explicit, but then maybe there's less of a point to having a `load` method.

Reply:
I don't have a strong opinion about how the output of "load(..)" is passed to "analyze(..)". I'm happy to change this.

Maybe enforce that `load(..)` returns a dict, which is passed as keyword args to `analyze(..)` like so:

```
input = EntityAnalyzer.load(event["transcript"], event["metadata"])
entities = EntityAnalyzer.analyze(**input)
```

where `analyze` is defined as:

```
def analyze(self, *, texts, event_time)
```

This makes it so that `analyze` doesn't take any positional arguments (except for the implicit `self`), enforcing that `texts` and `event_time` are passed as keyword args, which is accomplished using the spread operator?

If all of that's too complicated, I'm fine with passing a dict.

Reply:
I think the `load` function returning a `Dict[str, Any]` to be passed to the `analyze` function would be the best solution here. Still a single object to keep with @oliviagunton's layout of the module but we get those nice `**kwargs` style loading. Unless I misunderstood this conversation.